### ISSUE TITLE
Destructuring assignment causes eslint error

### ISSUE THREAD

**Issue by lll000111** [gajus/eslint-plugin-flowtype#295]
Demo code:

```js
// @flow
'use strict';

import type {MyObj} from './module';

// eslint-disable-next-line no-unused-vars
const {obj}: {obj: MyObj} = {
    obj: {foo: 42}
};
```

Error:

```
$ eslint .

C:\Projects\Test\src\main.js
  7:20  error  'MyObj' is already defined  no-redeclare

✖ 1 problem (1 error, 0 warnings)
```

eslintrc.js: https://pastebin.com/MCLULQJq

-------------

When I change the setup slightly I get eslint error "MyObj is defined but never used" &mdash; apart from the details (this one is harder to create a tiny demo for), how can this happen, eslint should not get to see any of those type strings?

**Comment by taion** [comment]
I'm seeing this too. I think this might be an issue when using babel-eslint 8.x.

**Comment by luetkemj** [comment]
also seeing this with `"babel-eslint": "8.2.2",` and I **do not** have `eslint-plugin-flowtype` installed
